modules/common/src/main/scala/emil/Access.scala

```diff
@@ -25,6 +25,10 @@ trait Access[F[_], C] {
 
   def getMessageCount(folder: MailFolder): MailOp[F, C, Int]
 
+  def getFolderNextUid(folder: MailFolder): MailOp[F, C, MailUid]
```

Comment:
I'm just a little concerned that imap stuff is creeping into the main api. I think this here is okish since it's only an id. And  it is very useful, there should be a way to have it included. If we want to wrap more imap-specific stuff, maybe a second trait would be good? Other backends can then just not implement it instead of throwing exceptions - wdyt?

I wanted to create a second backend for a long time, but tbh this might never happen… :)

Reply:
I've separated out the IMAP specific functionality.

Reply:
Oh wow… that is so nice - thank you! 💯 I'm sorry if I implied to do it right now, was actually just wanting to pass a thought. 😬